### PR TITLE
[Strangeness tracking] Enable 3body strangeness tracking by default

### PR DESCRIPTION
**Pull request by creetz16**
Dear @chiarazampolli and @shahor02 ,

Would it be possible to enable this feature (tracking of decay3bodys) in the strangeness tracker by default?
I tested the strangeness tracker with and without the feature enabled locally on some pp and PbPb CTFs to check the timing overhead. I got the following results:
-  **pp:** Test on two CTFs from Run 543113 (70063 events), I found a less than 3% increase in CPU time when adding the 3body tracking compared to without 
- **PbPb:** Test on 4 CTFs from Run 544122 (1330 events), I found that the timing overhead is negligible, the CPU times were comparable when running with and without the 3body tracking enabled

Could you let me know if anything else is needed?
I am also tagging @fmazzasc with whom I was discussing.

Thanks in advance! :)

### CODE REVIEW

**Comment by shahor02** [comment]
@creetz16 at the moment removed the draft tag to trigger CI. 
Fine for me but did you check also the AO2D volume change?

**Comment by creetz16** [comment]
Thanks @shahor02,
I checked the AO2D file sizes in the cases mentioned above and found that the increase is less than 0.1% both in PbPb and pp. 
I also included a small fix to enable the 3body tracking from the SVertexer code (see [c501ecc](https://github.com/AliceO2Group/AliceO2/pull/13182/commits/c501ecc5a972f4115693fec4cd25dc9ff9b356e0)).

**Comment by creetz16** [comment]
Hi @shahor02 did you have the chance to have a look at my previous comment?

File: Detectors/Vertexing/StrangenessTracking/src/StrangenessTracker.cxx [hunk 1]
```diff
@@ -242,58 +242,62 @@ void StrangenessTracker::processCascade(int iCasc, const Cascade& casc, const Ca
...
-      const auto& ITSindexRef = mSortedITSindexes[iTrack];
-      if (mStrParams->mVertexMatching && (mITSvtxBrackets[ITSindexRef].getMin() > dec3bodyIdx.getVertexID() || mITSvtxBrackets[ITSindexRef].getMax() < dec3bodyIdx.getVertexID())) {
-        continue;
-      }
-      if (matchDecayToITStrack(dec3bodyR, strangeTrack, structClus, itsTrack, daughterTracks, iThread)) {
-        auto propInstance = o2::base::Propagator::Instance();
-        o2::track::TrackParCov decayVtxTrackClone = strangeTrack.mMother; // clone track and propagate to decay vertex
-        if (!propInstance->propagateToX(decayVtxTrackClone, strangeTrack.mDecayVtx[0], getBz(), o2::base::PropagatorImpl<float>::MAX_SIN_PHI, o2::base::PropagatorImpl<float>::MAX_STEP, mCorrType)) {
-          LOG(debug) << "Mother propagation to decay vertex failed";
+  if (!mStrParams->mSkip3Body) {
```
Comment (shahor02) [inline Detectors/Vertexing/StrangenessTracking/src/StrangenessTracker.cxx:245]: Why do you need to check this here if it is already check before calling this method in https://github.com/AliceO2Group/AliceO2/pull/13182/files#diff-6c56393722a05e498b450726c3c3b759d0722698379743b271c92c5f82bf8197R319-R323 ?

File: Detectors/Vertexing/src/SVertexer.cxx [hunk 2]
```diff
@@ -1182,6 +1182,11 @@ int SVertexer::check3bodyDecays(const V0Index& v0Idx, const V0& v0, float rv0, s
       m3bodyTmp[ithread].push_back(candidate3B);
     }
     m3bodyIdxTmp[ithread].emplace_back(decay3bodyVtxID, v0Idx.getProngID(0), v0Idx.getProngID(1), bach.gid);
+
+    Decay3BodyIndex decay3bodyIdx(decay3bodyVtxID, v0Idx.getProngID(0), v0Idx.getProngID(1), bach.gid);
+    if (mStrTracker) {
+      mStrTracker->process3Body(m3bodyIdxTmp[ithread].size() - 1, candidate3B, decay3bodyIdx, ithread);
+    }
```
Comment (shahor02) [inline Detectors/Vertexing/src/SVertexer.cxx:1189]: And why do you need to call it here if it is already called from the `StrangenessTracker::process()` ?

**Review by shahor02 — COMMENTED** [review]
@creetz16 sorry, missed your reply. If the size and CPU overheads are so small, I see no reason to have it disabled, but I've got some questions on you PR, see below.

File: Detectors/Vertexing/StrangenessTracking/src/StrangenessTracker.cxx (hunk 1 above)
Comment (creetz16) [reply Detectors/Vertexing/StrangenessTracking/src/StrangenessTracker.cxx:245]: HI @shahor02 
As far as I understand the `StrangenessTracker::process()` is never called from within the SVertexer but only the individual `processV0`, `processCascade` and `process3Body` functions. This is why I followed the existing logic for V0s and cascades and added the check here in the SVertexer code. The check within the `StrangenessTracker::process()` allows to still run the StrangenessTracker as standalone task calling the `StrangenessTracker::process()` function.

File: Detectors/Vertexing/src/SVertexer.cxx (hunk 2 above)
Comment (creetz16) [reply Detectors/Vertexing/src/SVertexer.cxx:1189]: Same comment as above.

**Review by shahor02 — APPROVED** [review]
@creetz16 Indeed, I forgot that now the `process()` of the strangeness tracker in not used anymore unless it is run in a standalone mode. Sorry for that, merging.

**Comment by shahor02** [comment]
@creetz16 @fmazzasc if this should go to productions, please set relevant labels.

**Comment by creetz16** [comment]
+async-label async-2023-pbpb-apass4, async-2024-pp-apass1

**Pull request merged**